### PR TITLE
Rename formatter config and switch to an opt-out list

### PR DESCRIPTION
**Pull request by pugnascotia**
Move and rename formatter config file, so that it is easier for Eclipse users to import.

Also switch to an opt-out list for formatting. Instead of explcitly listing projects that should be formatted, instead list projects that should not be formatted. This means that any new projects will automatically be formatted and checked.

### CODE REVIEW

**Comment by pugnascotia** [comment]
@mark-vieira any suggestions for improving how the list of projects is embedded? It's not as long as I worried it would be, but I wondered about externalising it.

File: build.gradle [hunk 1]
```diff
@@ -105,25 +105,133 @@ subprojects {
...
-      ':x-pack:plugin:autoscaling',
-      ':x-pack:plugin:enrich',
-      ':x-pack:plugin:logstash'
+    // Projects that should be excluded from automatic formatting and
+    // checking with Spotless are listed here, by project path. Once the
+    // number of unformatted projects reaches zero, this mechanism will be
+    // removed.
+    //
+    // No new projects should be added here!
+    def projectPathsToExclude = [
```
Comment (jasontedor) [inline build.gradle]: I think this list will have to be regenerated on backport.

**Review by jasontedor — APPROVED** [review]
LGTM. I don't think we need to backport this to 7.6 though.

File: build.gradle (hunk 1 above)
Comment (pugnascotia) [reply build.gradle]: Yes, I was anticipating that.

**Comment by pugnascotia** [comment]
@elasticmachine run elasticsearch-ci/bwc 
@elasticmachine run elasticsearch-ci/default-distro

**Comment by pugnascotia** [comment]
@elasticmachine update branch

**Review by mark-vieira — APPROVED** [review]
I think it might be worth moving all this stuff into its own script plugin, just to reduce the amount of noise in the root build.gradle file. Perhaps create a `gradle/formatting.gradle` file and then just apply that in the root build script.

File: build.gradle [hunk 2]
```diff
@@ -63,6 +63,8 @@ if (VersionProperties.elasticsearch.toString().endsWith('-SNAPSHOT')) {
 }
 String elasticLicenseUrl = "https://raw.githubusercontent.com/elastic/elasticsearch/${licenseCommit}/licenses/ELASTIC-LICENSE.txt"
 
+apply from: 'gradle/formatting.gradle'
```
Comment (mark-vieira) [inline build.gradle]: nit: can we move this up to where we apply all the other script plugins?

**Review by mark-vieira — APPROVED** [review]

**Comment by pugnascotia** [comment]
Backported to `7.x in 24ef800b243, where I regenerated the sub-project list.

**Pull request merged**